ct/ct.c

```diff
@@ -167,7 +167,7 @@ waittest(Test *ts)
     if (pid == -1) {
         die(3, errno, "wait");
     }
-    killpg(pid, 9);
+    killpg(pid, SIGUSR2);
```

Comment:
I'd suggest `SIGTERM` here, since this is a pretty ordinary "kill the process, but let it do something before it exits" situation.

Reply:
Considering https://github.com/beanstalkd/beanstalkd/issues/424#issuecomment-509476476, maybe we could avoid making any patch to ct here. That might be preferable, since it'll make updating ct easier in the future.

Reply:
I have tried this approach with atexit, but it leads to some tests failed. These tests kill child-server in the middle of test and then the controlling wrapper thinks that since the test has received SIGTERM (15), it has failed:
```
ct/_ctcheck
................E...EE........................................

cttest_binlog_basic: error (signal 15)
start server port=55911 pid=86446
>20 put 0 0 100 0
>20
<20 INSERTED 1
start server port=55913 pid=86447
>22 delete 1
<22 DELETED

cttest_binlog_empty_exit: error (signal 15)
start server port=55921 pid=86455
start server port=55922 pid=86456
>25 put 0 0 0 0
>25
<25 INSERTED 1

cttest_binlog_read: error (signal 15)
start server port=55924 pid=86458
>25 use test
<25 USING test
>25 put 0 0 120 4
>25 test
<25 INSERTED 1
>25 put 0 0 120 4
>25 tes1
<25 INSERTED 2
>25 watch test
<25 WATCHING 2
>25 reserve
<25 RESERVED 1 4
<25 test
>25 release 1 1 1
<25 RELEASED
>25 reserve
<25 RESERVED 2 4
<25 tes1
>25 delete 2
<25 DELETED
start server port=55926 pid=86459
>27 watch test
<27 WATCHING 2
>27 reserve
<27 RESERVED 1 4
<27 test
>27 delete 1
<27 DELETED
>27 delete 2
<27 NOT_FOUND

0 failures; 3 errors.
make: *** [check] Error 1
lines: 62.3% (1392 out of 2236)
```

Alternatively,

We can skip everything in this patch if we change `ct.c` to `killpg(pid, SIGTERM)`. Or did the SIGKILL (9) exist because of cleanup reasons? Then I tried this, all tests were cleanup properly. Did you have something in mind when you used SIGKILL (9) in ct.c?

Reply:
> We can skip everything in this patch if we change ct.c to killpg(pid, SIGTERM). Or did the SIGKILL (9) exist because of cleanup reasons? Then I tried this, all tests were cleanup properly. Did you have something in mind when you used SIGKILL (9) in ct.c?

I take it back. It was the uncleaned cache of coverage data that made me to think that. It is not enough to change to SIGTERM.